src/adhocracy_core/adhocracy_core/scripts/auto_transition_process_workflow.py

```diff
@@ -0,0 +1,105 @@
+def auto_transition_process_workflow():  # pragma: no cover
+    """Automatically transition workflow states.
+
+    usage::
+
+        bin/auto_auto_transition_process_workflow etc/development.ini
+    """
+    docstring = inspect.getdoc(auto_transition_process_workflow)
+    parser = argparse.ArgumentParser(description=docstring)
+    parser.add_argument('ini_file',
+                        help='path to the adhocracy backend ini file')
+    args = parser.parse_args()
+    env = bootstrap(args.ini_file)
+    _auto_transition_process_workflow(env['root'], env['registry'])
+    transaction.commit()
+    env['closer']()
+
+
+def _auto_transition_process_workflow(context: IResource, registry: Registry):
+    catalogs = find_service(context, 'catalogs')
+    query = search_query._replace(interfaces=IProcess,
+                                  resolve=True
+                                  )
+    processes = catalogs.search(query).elements
+    for process in processes:
+        workflow_assignment = registry.content.get_sheet(process,
+                                                         IWorkflowAssignment)
+        if _workflow_auto_transition_enabled(registry, workflow_assignment) \
+            and _workflow_auto_transition_needed(process, registry,
+                                                 workflow_assignment):
+            _do_workflow_auto_transition(process, registry)
+
+
+def _workflow_auto_transition_enabled(
+        registry: Registry, workflow_assignment: IWorkflowAssignment):
+    workflow_name = workflow_assignment.get()['workflow']
+    workflow_meta = registry.content.workflows_meta[workflow_name]
+    return workflow_meta['auto_transition']
+
+
+def _workflow_auto_transition_needed(context: IResource, registry: Registry,
+                                     workflow_assignment: IWorkflowAssignment):
+    workflow = registry.content.get_workflow(context)
+    next_states = workflow.get_next_states(context,
+                                           create_fake_god_request(registry))
+    if len(next_states) == 1:
+        next_state = next_states[0]
+        end_date = _get_current_state_end_date(workflow_assignment, next_state)
+        if end_date:
+            return now() > end_date
+    return False
+
+
+def _get_current_state_end_date(workflow_assignment: IWorkflowAssignment,
+                                next_state: str):
+    current_state = workflow_assignment.get()['workflow_state']
+    state_data_list = workflow_assignment.get()['state_data']
+    current_state_data = None
+    next_state_data = None
+    for state_data in state_data_list:
+        if state_data['name'] == current_state:
+            current_state_data = state_data
+        if state_data['name'] == next_state:
+            next_state_data = state_data
+    if current_state_data:
+        if next_state_data and \
+                current_state_data['end_date'] != \
+                next_state_data['start_date']:
+            return None
```

Comment:
why does it return None when the end_date of the current state is not the same as the start_date of the current state? If the goal is to not trigger a switch to the next state because there is an inconsistency between the dates then the function name is misleading. Should the consistency of the date be enforced somewhere (workflow validation?) else instead of silently doing nothing here?